src/js/load-package.ts

```diff
@@ -80,11 +80,21 @@ const DEFAULT_CHANNEL = "default channel";
 // Regexp for validating package name and URI
 const package_uri_regexp = /^.*?([^\/]*)\.whl$/;
 
-function _uri_to_package_name(package_uri: string): string | undefined {
+type ParsedPackageData = {
+  distribution: string;
+  version: string;
+};
+
+function _uri_to_package_data(
+  package_uri: string,
+): ParsedPackageData | undefined {
   let match = package_uri_regexp.exec(package_uri);
   if (match) {
-    let wheel_name = match[1].toLowerCase();
-    return wheel_name.split("-").slice(0, -4).join("-");
+    let wheelName = match[1].toLowerCase().split("-");
+    return {
+      distribution: wheelName.slice(0, -4).join("-"),
```

Comment:
I was referring to the name convention of [PEP-491](https://peps.python.org/pep-0491/#file-name-convention) as you had mentioned. I suppose it may be more straightforward to just use a simpler name like `name`/`pkgname`.

This part was confusing to me because as far as I can tell, `wheelName.slice(0, -4).join("-")` is going to either:
- return the `{distribution}` part of the wheel name(according to PEP-491)
- return the `{distribution}-{version}` part of the wheel name if it contains a build tag

So I'm a bit lost here on why the original code didn't take `wheel_name.split("-")[0]` instead. I'm not sure what you mean by this returning the full filename as well, it looks like removing the `slice` would only end up being the full file name, because of the regex: `const package_uri_regexp = /^.*?([^\/]*)\.whl$/;`.

Reply:
> I was referring to the name convention of [PEP-491](https://peps.python.org/pep-0491/#file-name-convention) as you had mentioned. I suppose it may be more straightforward to just use a simpler name like name/pkgname.

Makes sense. Thanks!

Reply:
> So I'm a bit lost here on why the original code didn't take wheel_name.split("-")[0] instead. I'm not sure what you mean by this returning the full filename as well, it looks like removing the slice would only end up being the full file name, because of the regex: const package_uri_regexp = /^.*?([^\/]*)\.whl$/;.

Oh, sorry, I think I made a wrong comment. You are right, it is returning a pkg name not a full URL.

`wheelName.slice(0, -4).join("-")` About this code, IIRC, there was a wheel file that has an incorrect filename which does not follow PEP-491, such as `pkg-name-here-version-tag-...-.whl`. So I guess we needed to join the `-` to get the correct pkg name. 

But I guess we can just ignore those kind of ill-formatted wheel filenames.

Reply:
The test added in #295 is failing, which is I assume the case of incorrect filenames as you have mentioned. I'll remove the test if such cases are safe to no longer be accounted for, since it looks like the previous method could still possibly have problems with a wheel name that contains a build tag.